internal/storage/v1/elasticsearch/spanstore/from_domain.go

```diff
@@ -75,31 +62,23 @@ func (FromDomain) convertRefType(refType model.SpanRefType) dbmodel.ReferenceTyp
 	return dbmodel.ChildOf
 }
 
-func (fd FromDomain) convertKeyValuesString(keyValues model.KeyValues) ([]dbmodel.KeyValue, map[string]any) {
-	var tagsMap map[string]any
+func (fd FromDomain) convertKeyValues(keyValues model.KeyValues) []dbmodel.KeyValue {
 	var kvs []dbmodel.KeyValue
 	for _, kv := range keyValues {
```

Comment:
the size is known upfront now
```suggestion
	kvs := make([]dbmodel.KeyValue, 0, len(keyValues))
	for i := range keyValues {
		kvs[i] = fd.convertKeyValue(keyValues[i])
```